### ISSUE TITLE
__popcnt not declared with mingw-w64 in bitops.hpp (line_descriptor modul)

### ISSUE THREAD

**Issue by Vaniax** [Itseez/opencv_contrib#256]
Hi,

building with mingw-w64 leads to the error: 
```
__popcnt is not declared in this scope
```
The error seems to be in the `ifdef` block in `bitops.hpp`:

```
#include "precomp.hpp"
#ifdef _WIN32
# include <intrin.h>
# define popcnt __popcnt
# pragma warning( disable : 4267 )
#else
# define popcnt __builtin_popcount
#endif
```
`#define popcnt __builtin_popcount` after the last `#endif`solves the problem.

Possible solution:

```
#if defined _WIN32 && !defined __MINGW32__
# include <intrin.h>
# define popcnt __popcnt
# pragma warning( disable : 4267 )
#else
# define popcnt __builtin_popcount
#endif

```

**Comment by ianvanwijk** [comment]
I had the same problem. It is because I do not have Visual Studio installed.

This fixed the problem for me:

```
#ifdef _WIN32
// # include <intrin.h>
# define popcnt __builtin_popcount
// # pragma warning( disable : 4267 )
#else
# define popcnt __builtin_popcount
#endif
```

Or remove it in total and just use:
```
# define popcnt __builtin_popcount
```

**Comment by ianvanwijk** [comment]
```
#if _MSC_VER >= 1600
```

To check if using Visual Studio compiler:
https://msdn.microsoft.com/en-us/library/b0084kay.aspx

**Comment by mshabunin** [comment]
Should be fixed in #370
